modules/data/src/test/java/io/fluo/webindex/data/IndexIT.java

```diff
@@ -0,0 +1,354 @@
+        new TableInfo(cluster.getInstanceName(), cluster.getZooKeepers(), "root", "secret",
+            exportTable), 5);
+
+    FluoFactory.newAdmin(config).initialize(
+        new FluoAdmin.InitOpts().setClearTable(true).setClearZookeeper(true));
+
+    miniFluo = FluoFactory.newMiniFluo(config);
+  }
+
+
+  @After
+  public void tearDownFluo() throws Exception {
+    if (miniFluo != null) {
+      miniFluo.close();
+    }
+  }
+
+  public static Map<String, Page> readPages(File input) throws Exception {
+    Map<String, Page> pageMap = new HashMap<>();
+    ArchiveReader ar = WARCReaderFactory.get(input);
+    for (ArchiveRecord r : ar) {
+      Page p = ArchiveUtil.buildPage(r);
+      if (p.isEmpty() || p.getOutboundLinks().isEmpty()) {
+        continue;
+      }
+      pageMap.put(p.getUrl(), p);
+    }
+    ar.close();
+    return pageMap;
+  }
+
+  private void assertOutput(Collection<Page> pages, FluoClient client) throws Exception {
+    JavaRDD<Page> pagesRDD = sc.parallelize(new ArrayList<>(pages));
+    Assert.assertEquals(pages.size(), pagesRDD.count());
+
+    // Create expected output using spark
+    IndexStats stats = new IndexStats(sc);
+    JavaPairRDD<RowColumn, Bytes> linkIndex = IndexUtil.createLinkIndex(stats, pagesRDD);
+    List<Tuple2<RowColumn, Bytes>> linkIndexList = linkIndex.collect();
+    JavaPairRDD<RowColumn, Bytes> linkIndexNoRank = IndexUtil.filterRank(linkIndex);
+
+    // Compare against actual
+    boolean foundDiff = false;
+    foundDiff |= diffExportTable(linkIndexList);
+    foundDiff |= diffFluoTable(client, linkIndexNoRank.collect());
+    if (foundDiff) {
+      dump(client);
+      dumpExportTable();
+      print(linkIndexList);
+    }
+    Assert.assertFalse(foundDiff);
+  }
+
+  @Test
+  public void testIndexing() throws Exception {
+
+    Map<String, Page> pages = readPages(new File("src/test/resources/wat-18.warc"));
+
+    try (FluoClient client = FluoFactory.newClient(miniFluo.getClientConfiguration())) {
+
+      try (LoaderExecutor le = client.newLoaderExecutor()) {
+        for (Page page : pages.values()) {
+          log.debug("Loading page {} with {} links", page.getUrl(), page.getOutboundLinks().size());
+          le.execute(PageUpdate.updatePage(page));
+        }
+      }
+      miniFluo.waitForObservers();
+
+      assertOutput(pages.values(), client);
+
+      String deleteUrl = "http://1000games.me/games/gametion/";
+      log.info("Deleting page {}", deleteUrl);
+      try (LoaderExecutor le = client.newLoaderExecutor()) {
+        le.execute(PageUpdate.deletePage(deleteUrl));
+      }
+      miniFluo.waitForObservers();
+
+      pages.remove(deleteUrl);
+      assertOutput(pages.values(), client);
+
+      String updateUrl = "http://100zone.blogspot.com/2013/03/please-memp3-4shared.html";
+      Page updatePage = pages.get(updateUrl);
+      updatePage.addOutboundLink("http://example.com", "Example");
+      updatePage.removeOutboundLink("http://www.blogger.com");
```

Comment:
could assert something was removed (maybe have remove method return prev value... then can assert not null)... or make the remove method throw an exception if nothing is removed....